meson.build

```diff
@@ -1631,7 +1639,9 @@ exe = executable('systemd-analyze',
                                  libblkid],
                  install_rpath : rootlibexecdir,
                  install : true)
-public_programs += exe
+if exe.found()
+        public_programs += exe
+endif
```

Comment:
So I actually need to refactor this a bit, due to a meson bug it doesn't seem possible to check it in the for each loop for `public_programs`.

Reply:
So I've had to replace `if exe.found()` with `if gperf.found() and libcap.found() and libmount.found()` since meson executables don't have a `found()` method, this is a little tedious since it requires manually specifying any dependencies that can potentially be disablers but this seems to be the only way to prevent a disabler from getting passed to `public_programs`(at least this is currently fairly straight forward since virtually all of these `public_programs` depend on `libshared` in some way which seems to have a hard dependency on `libmount`, `gperf` and `libcap` although I'm not 100% sure on that). Using disablers is still helpful since it automatically eliminates the actual builds themselves, it's generally just post build operations that you need to manually check all the disabler dependencies.

Reply:
> It seems much more reasonable to simply do the check there, and not here and every time something is added to the list.

While that would be ideal it doesn't appear to be something supported by meson, marking this resolved for now since doing the check within the `foreach` loop is apparently not possible.